lib/utils/html_widgets.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:html/dom.dart' as dom;
+import 'package:noheva_api/noheva_api.dart';
+import 'package:noheva_visitor_ui/widgets/video_widget.dart';
+
+/// Html Widgets
+///
+/// This class is used to build custom widgets from HTML elements
+class HtmlWidgets {
+  /// Builds custom widget from HTML [element]
+  ///
+  /// Parses elements resource and event trigger data from [resources] and [eventTriggers]
+  /// TODO: Implement support for layouts default resources once backend communication is ready.
+  static Widget? buildCustomWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) =>
+      switch (element.localName) {
+        CustomHtmlWidgets.IMAGE =>
+          _buildCustomImageWidget(element, resources, eventTriggers),
+        CustomHtmlWidgets.BUTTON =>
+          _buildCustomButtonWidget(element, resources, eventTriggers),
+        CustomHtmlWidgets.VIDEO =>
+          _buildCustomVideoWidget(element, resources, eventTriggers),
+        _ => null
+      };
+
+  /// Builds custom Image widget from HTML [element] and applies [resources]
+  ///
+  /// TODO: Implement loading images from device memory
+  static Widget _buildCustomImageWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? foundResource = resources
+        .firstWhereOrNull((resource) => resource.id == element.innerHtml)
+        ?.data;
+
+    Widget imageWidget = Image.network(
+      foundResource ?? "",
+      width: size.width,
+      height: size.height,
+    );
+
+    if (foundEventTrigger != null) {
+      return GestureDetector(
+        onTap: _handleTapEvent(foundEventTrigger),
+        child: imageWidget,
+      );
+    }
+
+    return imageWidget;
+  }
+
+  /// Builds custom Button widget from HTML [element] and applies [resources]
+  static Widget _buildCustomButtonWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? foundResource = resources
+        .firstWhereOrNull((resource) => resource.id == element.innerHtml)
+        ?.data;
+
+    return ElevatedButton(
+        style: ButtonStyle(fixedSize: MaterialStateProperty.all(size)),
```

Comment:
Styles should probably be derived from HTML element (can also be fixed with TODO-comment)

Reply:
Resolved with TODO comment